atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -0,0 +1,262 @@
+        self.build_id = None
+
+    def get_output(self, metadatas):
+        """
+        Build the output entry of the metadata.
+
+        :return: list, containing dicts of partial metadata
+        """
+
+        for instance in metadatas['output']:
+            instance['buildroot_id'] = instance['arch'] + '-' + repr(instance['buildroot_id'])
+
+        return metadatas['output']
+
+    def get_buildroot(self, metadatas):
+        """
+        Build the buildroot entry of the metadata.
+
+        :return: list, containing dicts of partial metadata
+        """
+        for instance in metadatas['buildroots']:
+            instance['id'] = instance['container']['arch'] + '-' + repr(instance['id'])
+
+        return metadatas['buildroots']
+
+    def get_build(self, metadata):
+        start_time = int(atomic_reactor_start_time)
+
+        labels = df_parser(self.workflow.builder.df_path, workflow=self.workflow).labels
+
+        component = get_preferred_label(labels, 'com.redhat.component')
+        version = get_preferred_label(labels, 'version')
+        release = get_preferred_label(labels, 'release')
+
+        source = self.workflow.source
+        if not isinstance(source, GitSource):
+            raise RuntimeError('git source required')
+
+        extra = {'image': {'autorebuild': is_rebuild(self.workflow)}}
+        koji_task_id = metadata.get('labels', {}).get('koji-task-id')
+        if koji_task_id is not None:
+            self.log.info("build configuration created by Koji Task ID %s",
+                          koji_task_id)
+            try:
+                extra['container_koji_task_id'] = int(koji_task_id)
+            except ValueError:
+                self.log.error("invalid task ID %r", koji_task_id, exc_info=1)
+
+        fs_result = self.workflow.prebuild_results.get(AddFilesystemPlugin.key)
+        if fs_result is not None:
+            try:
+                fs_task_id = fs_result['filesystem-koji-task-id']
+            except KeyError:
+                self.log.error("%s: expected filesystem-koji-task-id in result",
+                               AddFilesystemPlugin.key)
+            else:
+                try:
+                    task_id = int(fs_task_id)
+                except ValueError:
+                    self.log.error("invalid task ID %r", fs_task_id, exc_info=1)
+                else:
+                    extra['filesystem_koji_task_id'] = task_id
+
+        help_result = self.workflow.prebuild_results.get(AddHelpPlugin.key)
+        if isinstance(help_result, dict) and 'help_file' in help_result and 'status' in help_result:
+            if help_result['status'] == AddHelpPlugin.NO_HELP_FILE_FOUND:
+                extra['image']['help'] = None
+            elif help_result['status'] == AddHelpPlugin.HELP_GENERATED:
+                extra['image']['help'] = help_result['help_file']
+            else:
+                self.log.error("Unknown result from add_help plugin: %s", help_result)
```

Comment:
Hmm, had a thought about this: I don't think we can store `None` in build annotations. So actually let's JSON-encode that value, e.g.

- "\"/path/to/help.1\"" for the case where there is a help.1 file
- "nil" for the case where there is not a help.1 file
- annotation not present for the case where the add_help plugin didn't run